### PR TITLE
module 10 complete

### PR DESCRIPTION
**Pull request by nrowbotham**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: todo_app/app.py
Comment (JackMead) [inline todo_app/app.py:64]: Nice, I wasn't able to avoid this. It would be good to also apply it to the delete/update/change due date methods too since we wouldn't want people messing with those either, and currently non-writers can still delete to-dos if they know what they're doing!
(Similarly every route below should have `@login_required` set, otherwise people can access them without a role at all!)

File: todo_app/app.py
Comment (JackMead) [inline todo_app/app.py:16]: Minor, but it could be nice to make this name an environment variable so you can change who has write access per environment

**Review by JackMeadDev — CHANGES_REQUESTED** [review]
Nice work, this basically works but there are a few things we should update before we merge!
* We should get the tests working - you've done the hard part in M9 getting Mongo playing ball so it would be a shame to lose these now (also, it's the last time you should find they break 🤞 )
  * For your unit tests, you'll just need to pass a `User` object through to the view_model, e.g. switch the ViewModel initialisation to:
```python
view_model = ViewModel(items, User("TestUser"))
```
  * For the integration test, we want to switch off the redirects/login protection. The easiest is probably:
    * Inside `test_app.py`, set the `LOGIN_DISABLED` environment variable before creating the app
```python
    os.environ["LOGIN_DISABLED"] = "True"
```
    * And then check it inside the `Config` class:
```python
    LOGIN_DISABLED = os.environ.get("LOGIN_DISABLED") == "True"
```
* We should update your Readme - mention the need to set up an OAuth app in Github etc.
* We should secure your other endpoints - you've proved you can with the write endpoint, but any user can currently delete to-dos if they can guess what they're doing

**Review by JackMeadDev — COMMENTED** [review]
Thanks for making those changes, I was able to run your app and the tests without any issues and I'm happy this is merged 😄

**Pull request merged**
